### PR TITLE
Reusable Common Link Variants

### PR DESCRIPTION
**Pull request by iethree**
### Description

We had a number of styled components that kept creating the same variants of our base link component over and over again. This is slow, annoying, and increases our bundle size.  Here, I extract the two most common link variant and use them where appropriate in the codebase.

The brand and boldBrand both have underline hover states.

Screen Shot 2023-04-26 at 3 04 55 PM

This also adds a storybook story.

- - -
This change is [Reviewable](https://reviewable.io/reviews/metabase/metabase/30404)

### CODE REVIEW

**Review by ranquild — APPROVED** [review]
@iethree Maybe `<Link variant="brand" />` etc?

**Comment by iethree** [comment]
> @iethree Maybe `<Link variant="brand" />` etc?

Good suggestion. this is updated 👍 

links

File: frontend/src/metabase/core/components/Link/Link.styled.tsx [hunk 1]
```diff
@@ -18,3 +19,20 @@ export const LinkRoot = styled(Link, {
 
   ${focusOutlineStyle("brand")};
 `;
+
+export const variants = {
```
Comment (ranquild) [inline frontend/src/metabase/core/components/Link/Link.styled.tsx:23]: This is fine for now but can become problematic if we need to add additional props to links. Wdyt about the following:

```
const LinkRoot = styled(Link)<LinkRootProps>`
  ...

  ${props => props.variant === "brand" && css`
    color: ${metabaseColor("brand")};
    &:hover {
      text-decoration: underline;
    }
  `}

  ${props => props.variant === "brandBold" && css`
     ...
  `}
`;
```

or

```
const LinkRoot = styled(Link)<LinkRootProps>`
  ...

  ${props => props.variant !== "default" && css`
    color: ${metabaseColor("brand")};
    &:hover {
      text-decoration: underline;
    }
  `}

  ${props => props.variant === "brandBold" && css`
     fontWeight: bold;
  `}
`;
```

**Review by ranquild — APPROVED** [review]

**Comment by kdoh** [comment]
Happy to see the variant approach ending up the winner here. I find it to be a bit easier to just grab the `<Link>` and try to use it and go off and refer to the docs if your editor isn't helping you out enough with types and props opposed to have to know the named imports.

This isn't a merge blocker on my end, but `brandBold` is an interesting thing that jumps out at me.  I get the want to encapsulate a  style that appears a couple spots as a prop, but can't we just nest `<b>` tags within a link (or use it inside of a text with a bold style set) and use to achieve this the few places that's a thing? To me a "brandBold" link isn't a named design system thing that we think about. It's too low level and doesn't have specific behavior attached to it.

**Review by kdoh — APPROVED** [review]

**Pull request merged**